src/Persistence/Array_/ArrayByRef.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace atk4\data\Persistence\Array_;
+
+/**
+ * Array wrapper allows to pass an array as a reference/object.
+ */
+class ArrayByRef implements \ArrayAccess, \Iterator
+{
+    /** @var array */
+    private $arr;
+    /** @var bool */
+    private $invalidNext = false;
+    /** @var bool */
+    protected $breakReferenceOnClone;
+
+    public function __construct(array &$array, bool $breakReferenceOnClone = true)
```

Comment:
> @mvorisek: Passing anything by reference is bad practise, also not advised to do so by Nikic in general.

Reply:
Yes! I so so strictly againt that I have already added to the PR desc:
> If rerence is not needed at all, we can remove the ArrayByRef class completely. @DarkSide666 wdyt?

so if ok, I will remove this totally, I can also fix the tests.